### PR TITLE
[DEV-599] - Deploy workflow

### PR DESCRIPTION
**Pull request by kin0992**
#### List of Changes

- Remove workflow that published static website to GitHub pages
- Create CD workflow to publish static website to S3 Bucket

#### Motivation and Context

We want to automate the deploy of new version of static website on the S3 Bucket

#### How Has This Been Tested?

Manually, looking on AWS dashboard, checking resources changed their statuses

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/cd.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+name: Deploy Static Website to S3
```
Comment (datalek) [inline .github/workflows/cd.yaml:1]: How about changing the name to `Deploy Website` and the name of the file to `deploy-website`?

```suggestion
name: Deploy Website
```

File: .github/workflows/cd.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+on:
+  pull_request:
+    branches: '**'
```
Comment (datalek) [inline .github/workflows/cd.yaml:5]: What do you think if we launch it manually until the preview mode is implemented?

File: .github/workflows/cd.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        run: npm run compile
+
+      - name: Build static website
+        env:
+          GITBOOK_API_KEY: ${{ secrets.GITBOOK_API_KEY }}
+        run: npm run build -w nextjs-website
+
+      - name: Export static website
+        env:
+          GITBOOK_API_KEY: ${{ secrets.GITBOOK_API_KEY }}
```
Comment (datalek) [inline .github/workflows/cd.yaml:43]: I'm not sure but I think that the `export` command doesn't need the `GITBOOK_API_KEY`

Comment (kin0992) [reply .github/workflows/cd.yaml]: You are right, the API key is required during the build phase, where we fetching content from GitBook. I've removed from this step.

File: .github/workflows/cd.yaml (hunk 2 above)
Comment (kin0992) [reply .github/workflows/cd.yaml]: What do you think if we have 2 triggers?
- one manually
- one when the a PR has been merged

**Pull request merged**